torch/csrc/jit/passes/alias_analysis.cpp

```diff
@@ -927,20 +931,53 @@ void AliasDb::move(Node* toMove, Node* movePoint, MoveSide moveSide) {
   }
 }
 
+c10::optional<const Node*> AliasDb::getLastWildcard() const {
+  auto it = std::max_element(
+      wildcardNodes_.cbegin(),
+      wildcardNodes_.cend(),
+      [this](const Node* a, const Node* b) { return isBeforeSameGraph(a, b); });
+  if (it != wildcardNodes_.end()) {
+    return *it;
+  } else {
+    return c10::nullopt;
+  }
+}
+
 bool AliasDb::hasUntrackedEffects(Node* node) const {
```

Comment:
What are "untracked" effects?

Reply:
it's documented in the header—if we write to an input alias or potentially to a wildcard.

Reply:
So untracked means that it can show up as a side-effect visible outside of the function, correct?

Reply:
Yes; it also means that it writes to something and the effect may be visible to a wildcard later (but we don’t know).